docs/pages/architecture/agents.mdx

```diff
@@ -0,0 +1,255 @@
+authenticated with mutual TLS. The Proxy Service generates a short-lived X.509
+certificate signed by the cluster's host CA, with the client's identity and
+routing information for an infrastructure resource encoded in it, and uses it to
+authenticate with the Teleport agent.
+
+The Proxy Service authenticates the connection and dispatches it to the
+appropriate agent instance based on the routing information encoded in the
+client certificate, over the reverse tunnel.
+
+<Details title="Connecting to agents without reverse tunnels">
+
+It is possible to join Teleport agents to a cluster [through the Teleport Auth
+Service](../agents/join-services-to-your-cluster/join-token.mdx#connecting-directly-to-the-auth-service.mdx).
+Once an agent joins a cluster through the Teleport Auth Service, the Teleport
+Proxy Service dials the agent directly, without creating a reverse tunnel. This
+mode supports the Teleport SSH Service, Teleport Desktop Service, and Teleport
+Kubernetes Service.
+
+In direct mode, SSH Service instances act like OpenSSH servers that only accept
+client SSH certificates. Users can connect to SSH servers through the Teleport
+Proxy Service as a jump-host or directly:
+
+![Standard Mode](../../img/architecture/ssh-direct-mode@1.2x.svg)
+
+Direct mode is designed for legacy use cases and only supports self-hosted
+Teleport clusters. We recommend joining agents through the Teleport Proxy
+Service and the reverse tunnel system unless this is not possible for your
+self-hosted Teleport deployment.
+
+</Details>
+
+## Connecting to infrastructure through Teleport agents
+
+Teleport users can access resources in their infrastructure with the `tsh`
+clientCLI, the Teleport Web UI, and Teleport Connect. All of these tools route
+traffic through agents, usually routed through the Teleport Proxy Service.
+
+### Communication between agents and infrastructure resources
+
+Teleport agents proxy user traffic to resources in your infrastructure. The
+method that agents use to connect to a resource typically depends on whether the
+resource is self hosted or managed by a cloud provider. 
+
+For self-hosted resources, operators distribute mutual TLS credentials to
+infrastructure resources, including a certificate signed by a Teleport
+certificate authority for validating the resource and a certificate authority
+certificate that the resource can use for validating Teleport-issued client
+certificates. When an agent connects to a resource, it retrieves a signed client
+certificate from the Teleport Auth Service.
+
+For resources managed by cloud providers, Teleport takes advantage of identity
+and access management (IAM) solutions available through the cloud provider. The
+agent accesses credentials to authenticate to the cloud provider, and has
+permissions to perform certain operations against the infrastructure resources
+it protects, e.g., managing a database or assuming another IAM role.
```

Comment:
These three paragraphs only apply to databases I believe. The mechanism is different for every other type.